test/files/jvm/t8786-sig.scala

```diff
@@ -0,0 +1,112 @@
+class A[U] {
+  @annotation.varargs def m1[T]          (a: T*): T = a.head
+  @annotation.varargs def m2[T <: AnyRef](a: T*): T = a.head
+  @annotation.varargs def m3[T <: AnyVal](a: T*): T = a.head
+  @annotation.varargs def m4[T <: Int]   (a: T*): T = a.head
+  @annotation.varargs def m5[T <: String](a: T*): T = a.head
+  @annotation.varargs def m6             (a: String*): String = a.head
+  @annotation.varargs def m7             (a: Int*): Int = a.head
+  @annotation.varargs def m8             (a: U*): U = a.head
+
+  def n1[T]          (a: Array[T]): T = a(0)
+  def n2[T <: AnyRef](a: Array[T]): T = a(0)
+  def n3[T <: AnyVal](a: Array[T]): T = a(0)
+  def n4[T <: Int]   (a: Array[T]): T = a(0)
+  def n5[T <: String](a: Array[T]): T = a(0)
+  def n6             (a: Array[String]): String = a(0)
+  def n7             (a: Array[Int]): Int = a(0)
+  def n8             (a: Array[U]): U = a(0)
+}
+
+object Test extends App {
+  val a = classOf[A[_]]
+
+  def sig   (method: String, tp: Class[_]) = a.getDeclaredMethod(method, tp).toString
+  def genSig(method: String, tp: Class[_]) = a.getDeclaredMethod(method, tp).toGenericString
+  def bound (method: String, tp: Class[_]) = {
+    val m = a.getDeclaredMethod(method, tp)
+    m.getGenericParameterTypes.apply(0) match {
+      case _: Class[_] => ""
+      case gat: java.lang.reflect.GenericArrayType =>
+        val compTp = gat.getGenericComponentType.asInstanceOf[java.lang.reflect.TypeVariable[_]]
+        compTp.getBounds.apply(0).toString
+    }
+  }
+
+  def check(a: String, b: String) = {
+    assert(a == b, s"found: $a\nexpected: $b")
+  }
+
+  val sq = classOf[Seq[_]]
+  val ob = classOf[Object]
+  val ao = classOf[Array[Object]]
+  val as = classOf[Array[String]]
+  val ai = classOf[Array[Int]]
+
+  check(sig("m1", sq)   , "public java.lang.Object A.m1(scala.collection.Seq)")
+  check(sig("m2", sq)   , "public java.lang.Object A.m2(scala.collection.Seq)")
+  check(sig("m3", sq)   , "public java.lang.Object A.m3(scala.collection.Seq)")
+  check(sig("m4", sq)   , "public int A.m4(scala.collection.Seq)")
+  check(sig("m5", sq)   , "public java.lang.String A.m5(scala.collection.Seq)")
+  check(sig("m6", sq)   , "public java.lang.String A.m6(scala.collection.Seq)")
+  check(sig("m7", sq)   , "public int A.m7(scala.collection.Seq)")
+  check(sig("m8", sq)   , "public java.lang.Object A.m8(scala.collection.Seq)")
+
+  check(genSig("m1", sq), "public <T> T A.m1(scala.collection.Seq<T>)")
+  check(genSig("m2", sq), "public <T> T A.m2(scala.collection.Seq<T>)")
+  check(genSig("m3", sq), "public <T> T A.m3(scala.collection.Seq<T>)")
+  check(genSig("m4", sq), "public <T> T A.m4(scala.collection.Seq<T>)") // wrong signature for T <: Int, SI-9846
```

Comment:
Thanks for noting this. It would be doubleplus clear if it read something like:
> // TODO: this is testing the (incorrect) status quo -- the signature should be `public int A.m7(scala.collection.Seq<java.lang.Object>)`

 (assuming that is indeed the correct signature :-))